cvmfs/magic_xattr.cc

```
@@ -492,33 +556,77 @@ static std::string ListProxy(download::DownloadManager *dm) {
       buf += proxy_chain[i][j].url;
       buf += "\n";
     }
+
+    if (buf.size() > BaseMagicXattr::kMaxCharsPerPage) {
+      result_pages->push_back(buf);
+      buf = "";
+    }
+  }
+
+  if (buf.size() > 0) {
+    result_pages->push_back(buf);
   }
```

If we have no proxies defined, we now return an error for the corresponding attributes instead of an empty string. I think  we should maintain the current behavior.

---

i added it. but in the end i dont think we will manage ever to get in this position, unless you have a broken config.

no proxy set in config:
```
attr -g proxy_list /cvmfs/my.test.repo
Attribute "proxy_list" had a 7 byte value for /cvmfs/my.test.repo:
DIRECT
```

set empty proxy
```
sudo cvmfs_talk proxy set ' '
lhcb.cern.ch:
OK


 $ attr -g proxy_list /cvmfs/lhcb.cern.ch/
Attribute "proxy_list" had a 120 byte value for /cvmfs/lhcb.cern.ch/:
http:// 
http://128.142.161.84:3126
http://128.142.248.156:3126
http://131.225.188.245:3126
http://131.225.188.246:3126
```